src/v/transform/api.cc

```diff
@@ -218,6 +219,24 @@ class proc_factory : public processor_factory {
 
 } // namespace
 
+class wrapped_service_reporter : public rpc::reporter {
+public:
+    explicit wrapped_service_reporter(ss::sharded<service>* service)
+      : _service(service) {}
+
+    ss::future<model::cluster_transform_report> compute_report() override {
+        if (!_service->local_is_initialized()) {
```

Comment:
😢 do we really have races where the service may not be setup?

Reply:
Yes, we start the RPC server before the transform subsystem. I'm not sure I really want to do the surgery to fix that here, should I file a followup ticket?

Reply:
Added a comment.

Reply:
thanks. yeh no need to fix it in this PR. longer term it makes sense to try to excise that sort of special casing whenever possible.